src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -214,18 +214,18 @@ static void Main()
             comp.VerifyDiagnostics();
             comp.VerifyIL("C.Main", @"
 {
-  // Code size       28 (0x1c)
-  .maxstack  3
+  // Code size       27 (0x1b)
+  .maxstack  2
   .locals init (System.ValueTuple<int, int> V_0) //x
-  IL_0000:  ldloca.s   V_0
-  IL_0002:  ldc.i4.1
-  IL_0003:  ldc.i4.2
-  IL_0004:  call       ""System.ValueTuple<int, int>..ctor(int, int)""
-  IL_0009:  ldloca.s   V_0
-  IL_000b:  constrained. ""System.ValueTuple<int, int>""
-  IL_0011:  callvirt   ""string object.ToString()""
-  IL_0016:  call       ""void System.Console.WriteLine(string)""
-  IL_001b:  ret
+  IL_0000:  ldc.i4.1
+  IL_0001:  ldc.i4.2
+  IL_0002:  newobj     ""System.ValueTuple<int, int>..ctor(int, int)""
```

Comment:
We should avoid copying except for rare cases where the difference is observable. (for example when we are in a try/catch etc... copying is required).
Codegen knows how and when to inline struct constructors. Possibly there is an extra conversion in the tree or something. - like for example instead of     ```local = new VT(..)``` it sees   ``` local = (cast) new VT()``` 

This applies to the other IL diffs below.